pymodbus/framer/rtu_framer.py

```diff
@@ -167,6 +174,67 @@ def check_frame(self):
             Log.debug("Frame advanced, resetting header!!")
             callback(result)  # defer or push to a thread?
 
+    async def asyncProcess(self, exec_request):
+        """Queue the request and start the worker task if needed."""
+        if self._queue is None:
+            self._queue = asyncio.Queue()
+        if self._worker is None:
+            self._worker = asyncio.create_task(self._worker_task())
+        future: asyncio.Future = asyncio.Future()
+        await self._queue.put((future, exec_request))
+        return await future
```

Comment:
This will slow down normal requests, since they will always go into the queue and wait for a task switch.

A better approach would be:
a) execute directly, if nothing is executing
b) if something is executing then put future in queue and await the newly created future
c) execute something
d) pop from queue and release future
e) return with response

That way you do not need the worker task.